src/cppcache/src/statistics/AtomicStatisticsImpl.cpp

```diff
@@ -330,7 +329,13 @@ double AtomicStatisticsImpl::_incDouble(int32_t offset, double delta) {
     throw IllegalArgumentException(s);
   }
 
-  return (doubleStorage[offset] += delta);
+  double expected = doubleStorage[offset];
+  double value;
+  do {
+    value = expected + delta;
+  } while (!doubleStorage[offset].compare_exchange_weak(expected, value));
```

Comment:
If compare_exchange_weak() fails, should expected be refreshed with doubleStorage[offset]?